app/services/device_cookie.rb

```diff
@@ -0,0 +1,9 @@
+class DeviceCookie
+  def self.check_for_new_device(cookies, user)
+    return unless user&.id
+    cookies[:device] && Device.find_by(
+      user_id: user.id,
+      cookie_uuid: cookies[:device],
+    )
+  end
```

Comment:
Do we use this for anything other than checking if it exists? (i.e. always followed by a `.nil?` or `.present?` check)

If so, maybe it'd be better to have this return a boolean indicating if it is a new device. The expected return value for a missing user is a little ambiguous... though I also wonder what's the situation where a user would actually be blank?

```suggestion
  def self.new_device?(cookies, user)
    !!cookies[:device] && user.devices.exists?(cookie_uuid: cookies[:device])
  end
```

Reply:
There's also a couple [device-related methods in the `User` class](https://github.com/18F/identity-idp/blob/f00bb67d61d288bec7274b0ce25e8a2be90d68dc/app/models/user.rb#L394-L401), so I'd wonder if it could make more sense to add the method there as opposed to creating a new class for this.

e.g. `user.new_device?(cookies[:device])`

Reply:
> Do we use this for anything other than checking if it exists? (i.e. always followed by a .nil? or .present? check)

It gets passed to create_event_for_existing_device.
https://github.com/18F/identity-idp/blob/f00bb67d61d288bec7274b0ce25e8a2be90d68dc/app/services/user_event_creator.rb#L61-L68